pkg/activator/net/revision_backends.go

```diff
@@ -447,7 +449,12 @@ func (rbm *revisionBackendsManager) endpointsUpdated(newObj interface{}) {
 	}
 	dests := endpointsToDests(endpoints, networking.ServicePortName(rw.protocol))
 	rbm.logger.Debugf("Updating Endpoints: %q (backends: %d)", revID.String(), len(dests))
-	rw.destsCh <- dests
+	// Don't block sending on rw.destsCh if we're terminating
```

Comment:
> This is not true, since in select channels are picked randomly. Thus if both are be satisfied, both can in theory trigger.

I don't see how picking select channel at random can cause a problem. If `done` is closed but `case rw.destsCh <- dests` was picked here, then that implies implies that `case x := <-rw.destsCh` was also picked in `revisionWatcher.run()`'s `select` and we didn't block the informer goroutine.

> 
> I would rewrite this differently
> 
> ```
> select {
> case done:
>   return
> default:
>  //
> }
> rw.destCh <- dests
> ```
> 
> And not add the new channel at all, since `destCh` cannot be closed unless done channel is closed.

I like the approach but I think this version is still racy: 
  1. say ctx isn't canceled yet, the select above picks the default case and the informer goroutine is preempted just before sending on rw.destCh
  2. ctx gets canceled, revisionWatcher.run() notices and returns closing rw.destCh
  3. informer's goroutine continues and sends on the closed rw.destCh causing panic

Is that scenario plausible?

Reply:
- It is not a problem per se, but the comment doesn't seem right.
- It is extremely unlikely, but to that point the same applies to your change? I.e. select picked sending, but before the actual sending code executed then preemption happened -- so the same pattern might ensue. 
Considering that this shutdown scenario I am less concerned about it. But if with my suggestion we get `go test -run=100 -race` pass without this problem -- I'll be satisfied enough.